pkg/virtctl/imageupload/imageupload.go

```diff
@@ -0,0 +1,304 @@
+package imageupload
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	uploadcdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/uploadcontroller/v1alpha1"
+	cdiClientset "kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+const (
+	uploadRequestAnnotation = "cdi.kubevirt.io/storage.upload.target"
+	podPhaseAnnotation      = "cdi.kubevirt.io/storage.pod.phase"
+
+	uploadPodWaitInterval = 1 * time.Second
+	uploadPodWaitTimeout  = 30 * time.Second
```

Comment:
If the importer pod wasn't already cached on the node, pulling in the container and starting it could hit this 30 second timeout under normal operation. 

Maybe we should consider two timeouts. One that ensures that the upload pod is at least being scheduled within a reasonable time, and another that waits a little longer for the pod to come online. 

uploadPodScheduledWaitTimeout = 30 *time.Second
uploadPodRunningWaitTimeout = 120 * time.Second

Reply:
yeah, something like.

Maybe I'm over thinking this though. A simplier alternative might be to just make the timeout an optional command line arg.